homeassistant/components/rainmachine/select.py

```diff
@@ -101,7 +102,7 @@ async def async_setup_entry(
     }
 
     async_add_entities(
-        entity_map[description.key](entry, data, description, hass.config.units.name)
+        entity_map[description.key](entry, data, description, hass.config.units)
```

Comment:
Checking the `name` of the unit system will stop work if we rename or add a new unit system.
It is more future-proof to check the `temperature_unit` of the unit system.